pandas/_libs/tslibs/offsets.pyx

```diff
@@ -194,6 +218,47 @@ def _validate_business_time(t_input):
     else:
         raise ValueError("time data must be string or datetime.time")
 
+
+# ---------------------------------------------------------------------
+# Constructor Helpers
+
+_rd_kwds = set([
+    'years', 'months', 'weeks', 'days',
+    'year', 'month', 'week', 'day', 'weekday',
+    'hour', 'minute', 'second', 'microsecond',
+    'nanosecond', 'nanoseconds',
+    'hours', 'minutes', 'seconds', 'milliseconds', 'microseconds'
+    ])
+
+def _determine_offset(kwds):
```

Comment:
At the moment this is a method of `DateOffset` that only gets called in `__init__`.